source/constructing-lattices.md

```diff
@@ -0,0 +1,306 @@
+        name: thingC
+    - item:             # This item contains an Element that is reversed.
+        direction: -1
+        Element:
+          ...
+    - item:             # This item contains a BeamLine repeated three times    
+        repetition: 3
+        BeamLine:
+          ...
+```
+
+%---------------------------------------------------------------------------------------------------
+(s:line.construction)=
+### Constructing a BeamLine `line` 
+
+A line item which is a lattice element can be specified by name if a lattice element
+of that name has been defined. Example:
+```{code} yaml
+Element: 
+  name: q1w
+  type: Quadrupole
+  ...
+
+BeamLine:
+  line:
+    - item: q1w       # Line item is element q1w.
+    - item:           # Line item is element q1w the same as the previous item.
+        Element:
+            inherit: q1w
+    ...
+```
+
+A line item which is a lattice element can also be specified by defining the lattice element
+"in place" in the line. Example:
+```{code} yaml
+BeamLine:
+  line:
+    - item:
+        Element:
+          name: octA
+          type: Octupole
+          Kn3L: 0.34
+          ...
+    ...
+```
+
+Similarly, a line item which is a BeamLine can either be referred to by name or can
+be defined "in place". Example:
+```{code} yaml
+BeamLine:
+  line:
+    - item: inj_line          # Refer by name to a previously defined BeamLine
+    - item:
+        BeamLine:             # Define a subline in place
+          line:
+           ...
+    ...
+```
+
+Restriction: Infinite recursion of sublines is not allowed. 
+For example, if BeamLine `B` is a subline of `A`, then BeamLine `A` may not be a subline of `B`.
+
+%---------------------------------------------------------------------------------------------------
+(s:repetition)=
+### Repetition
+
+For any line `item`, a `repetition` count component can be used to represent multiple copies
+of the item. Example:
+```{code} yaml
+BeamLine:
+  name: full_line
+  line:
+    - item:
+        name: short_line
+        repetition: 3
+```
+In this case, `short_line` is repeated three times when the BeamLine is expanded to form a lattice
+branch. For example, if `short_line` is defined by:
+```{code} yaml
+BeamLine:
+  name: short_line
+  line:
+    - item: A
+    - item: B
+    - item: C
+```
+then the expanded `full_line` will look like:
+```{code} yaml
+A, B, C, A, B, C, A, B, C
+```
+
+repetition counts can be negative. In this case, the elements are taken to occur in reverse order.
+Thus, in the above example, if the `repetition` count was `-3`, the expanded `full_line` will
+look like:
+```{code} yaml
+C, B, A, C, B, A, C, B, A
+```
+
+Notice that reverse order does not mean true [direction reversal](#s:ref.construct). 
+For elements that have longitudinal symmetry, this does not matter. 
+However, for example, for a `Bend` element, that is in a line with reversed order,
```

Comment:
Remove the comma between "element" and "that".

Reply:
Done.